src/main/java/com/anaplan/client/AnaplanService.java

```diff
@@ -0,0 +1,228 @@
+package com.anaplan.client;
+
+import com.anaplan.client.ex.ActionsNotFoundException;
+import com.anaplan.client.ex.ProcessesNotFoundException;
+import com.anaplan.client.ex.ServerFilesNotFoundException;
+import com.google.common.base.Strings;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Serializable;
+import java.net.URI;
+import java.net.URISyntaxException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Wrapper class of the Anaplan service providing the interfaces for a client to connect with
+ * Anaplan server.
+ */
+public class AnaplanService implements Serializable {
+
+  /** Enum class of the Anaplan internal function types. */
+  public enum FunctionType {
+    PROCESS,
+    ACTION
+  }
+
+  /** Interface for the CDF plugin that use Anaplan service. */
+  public interface AnaplanConfig {
+    String getServiceLocation();
+
+    String getAuthServiceLocation();
+
+    String getUsername();
+
+    String getPassword();
+  }
+
+  public static final String NAME_SERVICE_LOCATION = "serviceLocation";
+  public static final String NAME_AUTH_SERVICE_LOCATION = "authServiceLocation";
+  public static final String NAME_USERNAME = "username";
+  public static final String NAME_PASSWORD = "password";
+  public static final String NAME_WORKSPACE_ID = "workspaceId";
+  public static final String NAME_MODEL_ID = "modelId";
+  public static final String NAME_SERVER_FILE_NAME = "serverFileName";
+
+  private static final Logger LOG = LoggerFactory.getLogger(AnaplanService.class);
+
+  /**
+   * Sets metadata details for the connection with Anaplan server
+   *
+   * @param config the config instance carries metadata for the connection with Anaplan server
+   */
+  public static void setTaskMetadata(AnaplanConfig config) {
+    setAPIRoot(config.getServiceLocation(), config.getAuthServiceLocation());
+    setCredential(config.getUsername(), config.getPassword());
+  }
+
+  private static void setAPIRoot(String serviceLocation, String authServiceLocation) {
+    try {
+      Program.setServiceLocation(new URI(serviceLocation));
+      Program.setAuthServiceLocation(new URI(authServiceLocation));
+    } catch (URISyntaxException e) {
+      throw new RuntimeException(e.getMessage());
+    }
+  }
+
+  private static void setCredential(String username, String password) {
+    Program.setUsername(username);
+    Program.setPassphrase(password);
+  }
+
+  /** Closes the connection with Anaplan server */
+  public static void closeDown() {
+    Program.closeDown();
+    LOG.info("Anaplan connection is closed.");
+  }
+
+  /**
+   * Runs the Anaplan functions defined in the model
+   *
+   * @param workspaceId the ID of the target workspace
+   * @param modelId the ID of the target model
+   * @param functionId the ID of the function to be run
+   * @param type the type {@link FunctionType} of the function to be run
+   * @throws InterruptedException when errors happens during the function run
```

Comment:
javadoc should include that it throws ProcessesNotFoundException and ActionsNotFoundException

Reply:
Thanks! Updated!